### ISSUE TITLE
storage sink in stuck after s3 full-space error

### ISSUE THREAD

**Issue by AkiraXie** [pingcap/tiflow#8153]
### What did you do?

1. open storage sink , data flows to s3 
2. s3 is full-space 
3. clear s3 data

### What did you expect to see?

 storage sink checkpoint can advance after s3 data clear

### What did you see instead?

1. storage sink cannot advance ,and lag is increasing
![image](https://user-images.githubusercontent.com/50363732/215978816-7229db0b-7601-4658-abd3-4ad3d63ddd54.png)
2. remove and recreate changefeed , 2nd changefeed is in stuck also
![image](https://user-images.githubusercontent.com/50363732/215979210-20405505-13e8-48c9-ae56-8d3a4e30652c.png)
3. restart ticdc node , remove and recreate changefeed , 3rd changefeed is ok
![image](https://user-images.githubusercontent.com/50363732/215979370-66918740-8e0a-4489-b838-2a519035e2bb.png)
4. in grafana dashboard , 1st and 2nd changefeed is  abnormally visible
![img_v2_916fc712-4033-488b-a1e9-8720169bd47g](https://user-images.githubusercontent.com/50363732/215979834-17269b1b-029b-4fe6-b49f-aae82d7fcdb3.jpg)

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
/ # /cdc version
Release Version: v6.6.0-alpha
Git Commit Hash: 8ebfef02d94397cb2d7f1057d2212a984c49d68e
Git Branch: heads/refs/tags/v6.6.0-alpha
UTC Build Time: 2023-01-31 02:03:15
Go Version: go version go1.19.5 linux/amd64
Failpoint Build: false
```

**Comment by zhaoxinyu** [comment]
The monitoring abnormality will be fixed in later PR.

**Comment by AkiraXie** [comment]
later , cdc storage changefeed is still stuck and is not recoverable after s3 error ,so the issue needs reopen

**Comment by zhaoxinyu** [comment]
Addressed by #8268
